mypy/messages.py

```diff
@@ -1035,6 +1035,12 @@ def operator_method_signatures_overlap(
                       forward_method, self.format(forward_class)),
                   context)
 
+    def forbidden_inference_of_object(self, expr: Type, context: Context) -> None:
+        self.fail('Variable inferred to {}, which is forbidden.'.format(
```

Comment:
I think the error message should be just `Need type annotation for variable`. I think you can re-use the existing method for this error.

Reply:
Would there be any use to modify this method to add a "reason" field to perhaps make the error message more friendly?

Reply:
I think it may be only confusing. A better idea would be to add a note with a link to docs under such error explaining why annotations are needed. I think this idea already appeared in one of the issues. This can probably be done in a separate PR.